### PR TITLE
Fix AllocateRegularStaticBox for FOH

### PR DESCRIPTION
**Pull request by EgorBo**
No idea how that happened but apparently I forgot to revert this part in https://github.com/dotnet/runtime/pull/79188 in one of the intermediate commits when I was blaming this part for a flaky NRE that turned out to be a missing `GCPROTECT_BEGININTERIOR`.

So this fixes boxed statics to have the expected codegen:
```csharp
static DateTime MyField { get; set; }
```
```diff
; Assembly listing for method get_MyField():System.DateTime
-      48B8C81E802CFF010000 mov      rax, 0x1FF2C801EC8
-      488B00               mov      rax, gword ptr [rax]
-      488B4008             mov      rax, qword ptr [rax+08H]
+      48B8789AE568A4010000 mov      rax, 0x1A468E59A78
+      488B00               mov      rax, qword ptr [rax]
       C3                   ret 
-; Total bytes of code 18
+; Total bytes of code 14
```
Will try stress jobs once innerloop completes

### CODE REVIEW

**Review by jkotas — APPROVED** [review]

**Review by jkotas — CHANGES_REQUESTED** [review]
Looks like the test failures are related.

**Comment by EgorBo** [comment]
Right, it seems to surface an unrelated bug introduced recently. Most likely in https://github.com/dotnet/runtime/pull/80670

File: src/coreclr/jit/morphblock.cpp [hunk 1]
```diff
@@ -387,6 +387,11 @@ GenTree* MorphInitBlockHelper::MorphCommaBlock(Compiler* comp, GenTreeOp* firstC
 
     GenTree* effectiveVal = lastComma->gtGetOp2();
 
+    if (effectiveVal->IsVectorConst())
```
Comment (tannergooding) [inline src/coreclr/jit/morphblock.cpp:390]: Should this instead be a check for `!effectiveVal->OperIs(...)` where `...` is anything that will hit `unreached` for `GT_ADDR`?

This is hitting `GT_CNS_VEC` today, but we could also hit it for a `GT_HWINTRINSIC` or another scenario, right?

**Review by tannergooding — APPROVED** [review]
JIT side change LGTM

**Review by jkotas — APPROVED** [review]

File: src/coreclr/jit/morphblock.cpp [hunk 2]
```diff
@@ -387,6 +387,11 @@ GenTree* MorphInitBlockHelper::MorphCommaBlock(Compiler* comp, GenTreeOp* firstC
 
     GenTree* effectiveVal = lastComma->gtGetOp2();
 
+    if (!effectiveVal->isIndir() && !effectiveVal->IsLocal())
```
[the remark this answers is not in the document]
Comment (EgorBo) [reply src/coreclr/jit/morphblock.cpp]: Ah, right. The naming is a bit confusing 🙂

**Review by SingleAccretion — APPROVED** [review]

**Pull request merged**